### PR TITLE
ChunkParserK + ChunkParserK.Generic + ParserK.Generic

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: docs/User/Project/Changelog.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
   `ParallelT`, `AheadT` for GHC versions 9.6 onwards. This is due to a
   breaking change in `transformers` 0.6. You can replace `lift` with
   `fromEffect` when using these as top level monads in a monad stack.
+* Streamly.Data.ParserK is renamed to Streamly.Data.ParserK.Chunked
```
Comment (harendra-kumar) [inline docs/User/Project/Changelog.md:13]: You can also write how to adapt existing code to this change.

**Review by harendra-kumar — APPROVED** [review]

**Comment by harendra-kumar** [comment]
Let's name the module `Streamly.Data.ChunkParserK`. That way if we need to use a boxed array buffer, we can use the `.Generic` suffix.

File: core/src/Streamly/Data/ChunkParserK.hs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+-- |
```
[the remark this answers is not in the document]
Comment (adithyaov) [reply core/src/Streamly/Data/ChunkParserK.hs:1]: Ah, I remember moving it. I'll do it again.

File: core/src/Streamly/Data/ChunkParserK.hs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import Streamly.Internal.Data.Unbox (Unbox)
+import qualified Streamly.Internal.Data.Parser as ParserD
+
+import Streamly.Internal.Data.ChunkParserK
+
+-- | Convert a 'Fold' to a 'ChunkParserK'.
+--
+{-# INLINE fromFold #-}
+fromFold :: (MonadIO m, Unbox a) => Fold m a b -> ChunkParserK a m b
+fromFold = fromParser . ParserD.fromFold
```
[the remark this answers is not in the document]
Comment (adithyaov) [reply core/src/Streamly/Data/ChunkParserK.hs:49]: We can, I did not change anything, kept things as they were.

File: core/src/Streamly/Data/ParserK.hs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    -- ** Conversions
+    , fromFold
+    , fromParser
+    -- , toParser
+
+    -- ** Without Input
+    , fromPure
+    , fromEffect
+    , die
+    )
```
[the remark this answers is not in the document]
Comment (adithyaov) [reply core/src/Streamly/Data/ParserK.hs]: Yeah, could do that. And also export `ParserK`

**Comment by adithyaov** [comment]
See https://github.com/composewell/streamly/pull/2502

**Pull request closed without merging**